Add DNS provider for Scaleway

vanonox opened the pull request:
This PR adds support for Scaleway DNS : https://developers.scaleway.com/en/
Disclaimer: I work for Scaleway

it is to replace this PR https://github.com/go-acme/lego/pull/723#issuecomment-471175636

this is the output of the command to test : 
```
./lego -m vanonox@gmail.com --dns scaleway -d *.vanhau.net -d vanhau.net -s https://acme-staging-v02.api.letsencrypt.org/directory run
2020/01/23 13:21:17 Please review the TOS at https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017.pdf
Do you accept the TOS? Y/n
Y
2020/01/23 13:21:19 [INFO] acme: Registering account for vanonox@gmail.com
!!!! HEADS UP !!!!

		Your account credentials have been saved in your Let's Encrypt
		configuration directory at "/Users/vanono/src/github.com/vanonox/lego/dist/.lego/accounts".
		You should make a secure backup	of this folder now. This
		configuration directory will also contain certificates and
		private keys obtained from Let's Encrypt so making regular
		backups of this folder is ideal.2020/01/23 13:21:20 [INFO] [*.vanhau.net, vanhau.net] acme: Obtaining bundled SAN certificate
2020/01/23 13:21:21 [INFO] [*.vanhau.net] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/34550494
2020/01/23 13:21:21 [INFO] [vanhau.net] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/34550495
2020/01/23 13:21:21 [INFO] [*.vanhau.net] acme: use dns-01 solver
2020/01/23 13:21:21 [INFO] [vanhau.net] acme: Could not find solver for: tls-alpn-01
2020/01/23 13:21:21 [INFO] [vanhau.net] acme: Could not find solver for: http-01
2020/01/23 13:21:21 [INFO] [vanhau.net] acme: use dns-01 solver
2020/01/23 13:21:21 [INFO] [*.vanhau.net] acme: Preparing to solve DNS-01
2020/01/23 13:21:21 [INFO] [vanhau.net] acme: Preparing to solve DNS-01
2020/01/23 13:21:21 [INFO] [*.vanhau.net] acme: Trying to solve DNS-01
2020/01/23 13:21:21 [INFO] [*.vanhau.net] acme: Checking DNS record propagation using [9.9.9.9:53]
2020/01/23 13:21:21 [INFO] Wait for propagation [timeout: 2m0s, interval: 10s]
2020/01/23 13:21:22 [INFO] [*.vanhau.net] acme: Waiting for DNS record propagation.
2020/01/23 13:21:32 [INFO] [*.vanhau.net] acme: Waiting for DNS record propagation.
2020/01/23 13:21:42 [INFO] [*.vanhau.net] acme: Waiting for DNS record propagation.
2020/01/23 13:21:58 [INFO] [*.vanhau.net] The server validated our request
2020/01/23 13:21:58 [INFO] [vanhau.net] acme: Trying to solve DNS-01
2020/01/23 13:21:58 [INFO] [vanhau.net] acme: Checking DNS record propagation using [9.9.9.9:53]
2020/01/23 13:21:58 [INFO] Wait for propagation [timeout: 2m0s, interval: 10s]
2020/01/23 13:22:06 [INFO] [vanhau.net] The server validated our request
2020/01/23 13:22:06 [INFO] [*.vanhau.net] acme: Cleaning DNS-01 challenge
2020/01/23 13:22:06 [INFO] [vanhau.net] acme: Cleaning DNS-01 challenge
2020/01/23 13:22:06 [INFO] [*.vanhau.net, vanhau.net] acme: Validations succeeded; requesting certificates
2020/01/23 13:22:07 [INFO] [*.vanhau.net] Server responded with a certificate.
```

ldez commented:
Hello, in order for a PR adding a DNS provider to be accepted, you have to:

- [ ] add a description to your PR
- [ ] be able to maintain this provider
- [ ] have a homogeneous design with the other providers
- [ ] add tests (units)
```shell
make test
```
- [ ] add tests ("live")
https://github.com/go-acme/lego/blob/c938de68f262c057542d44bcb67ef961cefcc60e/providers/dns/cloudflare/cloudflare_test.go#L125-L151

```shell
make test
```
- [ ] add the provider to the [readme.md](https://github.com/go-acme/lego#dns-providers)
- [ ] add a [provider descriptor](https://github.com/go-acme/lego/blob/master/providers/dns/cloudflare/cloudflare.toml)
- [ ] generate CLI help and documentation
```bash
make generate-dns
```
- [ ] be able to do: _(and put the output of this command to a comment in your PR)_
```bash
rm -rf .lego

./lego -m your@email.com --dns YOUR_PROVIDER_NAME -d *.example.com -d example.com -s https://acme-staging-v02.api.letsencrypt.org/directory run
```
Note the wildcard domain is important.
- [ ] pass the linter ([golangci-lint](https://github.com/golangci/golangci-lint#install) must be installed):
```shell
make checks
```
- [ ] do `go mod tidy`

vanonox commented:
hi @ldez , i have added scaleway in the readme.md
`generate CLI help and documentation`i have added the files in the PR (thought it was always regenerated)
